nixos/modules/programs/obs-studio.nix

```diff
@@ -39,11 +39,16 @@ in
   };
 
   config = lib.mkIf cfg.enable {
-    programs.obs-studio.finalPackage = pkgs.wrapOBS.override { obs-studio = cfg.package; } {
-      plugins = cfg.plugins;
+    assertions = lib.singleton {
+      assertion = cfg.package == null -> cfg.plugins == [ ];
+      message = "Plugins cannot be set if package is null";
     };
 
-    environment.systemPackages = [ cfg.finalPackage ];
+    programs.obs-studio.finalPackage = lib.mapNullable
+      (obs-studio: pkgs.wrapOBS.override { inherit obs-studio; } { plugins = cfg.plugins; })
+      cfg.package;
+
+    environment.systemPackages = if cfg.finalPackage != null then [ cfg.finalPackage ] else [ ];
```

Comment:
I'm not sure how, but there has to be a cleaner way of doing this null logic. I thought there was a function `nullOr t -> [t]`, but I couldn't find it.